modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ComponentModel.java

```diff
@@ -175,6 +178,14 @@ public boolean isParameterValueProvidedBySchema(String parameterName) {
     return this.schemaValueParameter.contains(parameterName);
   }
 
+  public void addAnnotation(String annotationKey, Object value) {
```

Comment:
Perhaps I'm missing something here but, what would be the meaning of an annotation on a ComponentModel?